src/sentry/static/sentry/app/views/settings/projectSourceMaps/detail/index.tsx

```diff
@@ -0,0 +1,210 @@
+import React from 'react';
+import {RouteComponentProps} from 'react-router/lib/Router';
+import styled from '@emotion/styled';
+
+import {t} from 'app/locale';
+import AsyncView from 'app/views/asyncView';
+import {Organization, Project, Artifact} from 'app/types';
+import routeTitleGen from 'app/utils/routeTitle';
+import SearchBar from 'app/components/searchBar';
+import Pagination from 'app/components/pagination';
+import {PanelTable} from 'app/components/panels';
+import space from 'app/styles/space';
+import {formatVersion} from 'app/utils/formatters';
+import TextBlock from 'app/views/settings/components/text/textBlock';
+import SettingsPageHeader from 'app/views/settings/components/settingsPageHeader';
+import Button from 'app/components/button';
+import ButtonBar from 'app/components/buttonBar';
+import {IconReleases, IconChevron} from 'app/icons';
+import {
+  addErrorMessage,
+  addLoadingMessage,
+  addSuccessMessage,
+} from 'app/actionCreators/indicator';
+
+import SourceMapsArtifactRow from './sourceMapsArtifactRow';
+
+type RouteParams = {orgId: string; projectId: string; version: string};
+
+type Props = RouteComponentProps<RouteParams, {}> & {
+  organization: Organization;
+  project: Project;
+};
+
+type State = AsyncView['state'] & {
+  artifacts: Artifact[];
+};
+
+class ProjectSourceMaps extends AsyncView<Props, State> {
+  getTitle() {
+    const {projectId, version} = this.props.params;
+
+    return routeTitleGen(t('Source Maps %s', formatVersion(version)), projectId, false);
+  }
+
+  getDefaultState(): State {
+    return {
+      ...super.getDefaultState(),
+      artifacts: [],
+    };
+  }
+
+  getEndpoints() {
+    const {location} = this.props;
+
+    const endpoints: ReturnType<AsyncView['getEndpoints']> = [
+      ['artifacts', this.getArtifactsUrl(), {query: {query: location.query.query}}],
+    ];
+
+    return endpoints;
+  }
+
+  getArtifactsUrl() {
+    const {orgId, projectId, version} = this.props.params;
+
+    return `/projects/${orgId}/${projectId}/releases/${encodeURIComponent(
+      version
+    )}/files/`;
+  }
+
+  handleSearch = (query: string) => {
+    const {location, router} = this.props;
+
+    router.push({
+      ...location,
+      query: {...location.query, cursor: undefined, query},
+    });
+  };
+
+  handleDelete = async (id: string) => {
+    addLoadingMessage(t('Removing artifact\u2026'));
+
+    try {
+      await this.api.requestPromise(`${this.getArtifactsUrl()}${id}/`, {
+        method: 'DELETE',
+      });
+      // We might want to refetch the artifacts here to not mess up pagination
+      this.setState(state => ({artifacts: state.artifacts.filter(a => a.id !== id)}));
```

Comment:
You could re-fetch the current page. What would happen when you're on a page with a single item and that item is deleted? While unlikely it could come up.

Reply:
I just replaced it with `this.fetchData();`, might be a bit slower experience when deleting multiple items in a row, but we get better data reliability. And we will soon implement the deletion of entire archive at once anyway.